internal/maniffests/osbuild.go

```diff
@@ -0,0 +1,57 @@
+package maniffests
+
+import (
+	"context"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/go-logr/logr"
+	osbuilderprojectflottaiov1alpha1 "github.com/project-flotta/osbuild-operator/api/v1alpha1"
+	"github.com/project-flotta/osbuild-operator/internal/repository/osbuild"
+	"github.com/project-flotta/osbuild-operator/internal/repository/osbuildconfig"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+func CreateNewOSBuildCR(ctx context.Context, osBuildConfig *osbuilderprojectflottaiov1alpha1.OSBuildConfig,
+	logger logr.Logger, osBuildConfigRepository osbuildconfig.Repository, osBuildRepository osbuild.Repository,
+	scheme *runtime.Scheme) error {
+	osBuildNewVersion := *osBuildConfig.Status.LastVersion + 1
```

Comment:
I know you copied this code, but it just occurred to me that the first time you create an `OSBuild` CR, `LastVersion` will be `nil`. So, you need to check it